### PR TITLE
Ch2 - Mining: Part 2 of mining challenge more realistic

### PR DESCRIPTION
**Pull request by vinayaksh42**
Fixes: #488 

The user will now be able to find 1 block per 15 second (the same rate used in part-1 of the challenge). The 'transactions confirmed' as well as 'bitcoin earned' is updated along with 'Blocks found'.

preview: https://saving-satoshi-git-fork-vinayaksh42-issue488-savingsatoshi.vercel.app/en/chapters/chapter-2/mining-1?dev=true

### CODE REVIEW

**Comment by vinayaksh42** [comment]
Currently there is a bug with 10x mining. I am working on it

**Comment by ecurrencyhodler** [comment]
Tested it. 

If you set it to 15, the 10x appears before a block is found. I think the increment should be around 9 seconds. That's 1-2 seconds before the 10x button appears.

https://github.com/saving-satoshi/saving-satoshi/assets/32662508/c85b4dce-4e68-4a0a-91ae-aace718f4b87

**Comment by vinayaksh42** [comment]
> Tested it.
> 
> If you set it to 15, the 10x appears before a block is found. I think the increment should be around 8 seconds. That's 1-2 seconds before the 10x button appears.

I have updated the timing to 8second

**Comment by ecurrencyhodler** [comment]
tACK!
https://github.com/saving-satoshi/saving-satoshi/assets/32662508/7b77bab5-ebc8-437b-8a49-20000efce45c

**Review by shaavan — COMMENTED** [review]
I noticed a bug while testing the lesson.

After the first _four_ blocks are mined without pressing the `10x` button, the hash counter gets stuck on the "correct" hash, restarting again when the _fifth_ block is mined.

https://github.com/saving-satoshi/saving-satoshi/assets/85434418/1aa0a9aa-de17-442d-9a4d-aa8661a6a103

File: ui/common/MiningStatisticsHash.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ export default function MiningStatisticHash({
     }
 
     if (!hasherState && step === 2) {
-      const time = 1 * 15 * 1000
+      const time = 1 * 10 * 1000
```
Comment (afterburn) [inline ui/common/MiningStatisticsHash.tsx:82]: Can just be `10 * 1000`, multipying by `1` doesn't do anything.

File: content/lessons/chapter-2/mining-1.tsx [hunk 2]
```diff
@@ -89,6 +89,29 @@ export default function Mining1({ lang }) {
...
+  useEffect(() => {
+    let interval: NodeJS.Timeout
+    let currentBlock = blocks
+    if (ramdomNonce) {
+      interval = setInterval(() => {
+        currentBlock = currentBlock + 1
+        setBlocks(currentBlock)
+        setTransactionsConfirmed(currentBlock * 3500)
+        setBitcoinMined(currentBlock * 0.0061)
+      }, 1 * 8 * 1000)
```
Comment (afterburn) [inline content/lessons/chapter-2/mining-1.tsx:101]: Same here, `8 * 1000`.

**Review by afterburn — CHANGES_REQUESTED** [review]
Thanks @vinayaksh42, left some comments.

**Review by afterburn — APPROVED** [review]

**Pull request merged**